packages/suite/src/components/suite/ChangeDeviceLabel.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import styled, { css } from 'styled-components';
+
+import { Button, INPUT_HEIGHTS, Input } from '@trezor/components';
+import { Translation } from 'src/components/suite';
+import { useTranslation } from 'src/hooks/suite';
+import { MAX_LABEL_LENGTH } from 'src/constants/suite/device';
+import { SCREEN_SIZE } from '@trezor/components/src/config/variables';
+import { isAscii } from '@trezor/utils';
+
+interface ChangeDeviceLabelProps {
+    label: string;
+    error: string | null;
+    isButtonDisabled: boolean;
+    isInputDisabled?: boolean;
+    placeholder?: string;
+    isVertical?: boolean;
+    setLabel: (value: string) => void;
+    setError: (value: string | null) => void;
+    handleOnClick: () => void;
+}
+
+const StyledInput = styled(Input)<{ isVertical?: boolean }>`
+    ${props =>
+        props.isVertical &&
+        css`
+            margin: 4px 0 4px 4px;
+
+            &:not(:first-child) {
+                margin-left: 8px;
+            }
+
+            @media (max-width: ${SCREEN_SIZE.SM}) {
+                margin: 0px 0 10px;
+                width: 100%;
+            }
+        `}
+`;
+
+const StyledButton = styled(Button)<{ isVertical?: boolean; isButtonDisabled?: boolean }>`
+    height: ${INPUT_HEIGHTS.large}px;
+
+    ${props =>
+        props.isVertical &&
+        css`
+            min-width: 170px;
+            margin: 4px 0 4px 4px;
+            cursor: ${props.isDisabled ? 'not-allowed' : 'pointer'};
+            &:not(:first-child) {
+                margin-left: 8px;
+
+                @media (max-width: ${SCREEN_SIZE.SM}) {
+                    margin-left: 0;
+                }
+            }
+
+            @media (max-width: ${SCREEN_SIZE.SM}) {
+                width: 100%;
+                margin: 0;
+            }
+        `}
+`;
+
+export const ChangeDeviceLabel = ({
+    label,
+    error,
+    isButtonDisabled,
+    isInputDisabled,
```

Comment:
I know it was here originally, but I don't think we need to disable the input. The `isLocked` state is usually very short-lived. You can remove it so that it works the same in both instances.